### DISCUSSION TITLE
Make space behind `->` optional (for due dates)

### DISCUSSION THREAD

**Discussion by jotaen** [jotaen/xit#37] [File Format]
This is a topic that has been brought up in https://github.com/jotaen/xit/discussions/3.

Currently, the [spec requires due dates to be preceded by `-> ` (with trailing space)](https://github.com/jotaen/xit/blob/main/Specification.md#due-date), e.g. `-> 2022-05-01`. Without that trailing space, the due date technically is not recognised as such (e.g. `->2022-05-01`).

However, one could argue that the difference between `-> 2022-05-01` (with space) and `->2022-05-01` (without space) is rather subtle, and it might be more user-friendly to allow both variants to appear.

The downside of making the space optional is that it adds another syntax variant, which (theoretically) increases the complexity of the file format. This is not so much about this one particular case, but it’s more a general and overall topic. In the end, I think it comes down to a cost-benefit question.

**Comment by majkinetor** [comment]
Its better IMO to fist resolve https://github.com/jotaen/xit/discussions/3 and this dilemma might not even exist at the end.

Anyhow, I believe syntax variant just adds complexity, will be harder to parse, wont behave atomically in editor etc.
